src/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str_str.cs

```diff
@@ -639,6 +639,13 @@ public void SearchPatternLongPath()
 
             if (TestFiles)
             {
+                if (PlatformDetection.IsNetNative)
+                {
+                    // UapAot throws if the Directory of the file to create doesn't exist.
+                    Assert.Throws<DirectoryNotFoundException>(() => File.Create(longFullname));
```

Comment:
Compare also https://github.com/dotnet/corefx/pull/22513 which now I'm unsure about. In that case, I see Core and Desktop both throw PTLE. According to the change, .NET Native gives DNFE. I'm not sure when IOException happens (which is a base class of both of course)